src/renderer/kube-object.store.ts

```diff
@@ -71,42 +77,51 @@ export abstract class KubeObjectStore<T extends KubeObject = any> extends ItemSt
     }
   }
 
-  protected async loadItems(allowedNamespaces?: string[]): Promise<T[]> {
-    if (!this.api.isNamespaced || !allowedNamespaces) {
-      return this.api.list({}, this.query);
-    } else {
-      return Promise
-        .all(allowedNamespaces.map(namespace => this.api.list({ namespace })))
-        .then(items => items.flat());
+  protected async loadItems({ isAdmin, namespaces, api }: KubeObjectStoreLoadingParams): Promise<T[]> {
+    if (!api.isNamespaced) {
+      if (isAdmin) return api.list({}, this.query);
+
+      return [];
     }
+
+    return Promise
+      .all(namespaces.map(namespace => api.list({ namespace })))
+      .then(items => items.flat());
   }
 
   protected filterItemsOnLoad(items: T[]) {
     return items;
   }
 
   @action
-  async loadAll() {
+  async loadAll(params: { namespaces?: string[] } = {}) {
     this.isLoading = true;
     let items: T[];
 
     try {
-      const { allowedNamespaces, accessibleNamespaces, isAdmin } = getHostedCluster();
+      let contextNamespaces = params.namespaces;
+
+      if (!params.namespaces) {
+        const { namespaceStore } = await import("./components/+namespaces/namespace.store");
 
-      if (isAdmin && accessibleNamespaces.length == 0) {
-        items = await this.loadItems();
-      } else {
-        items = await this.loadItems(allowedNamespaces);
+        await namespaceStore.whenReady;
+        contextNamespaces = namespaceStore.getContextNamespaces();
       }
 
+      items = await this.loadItems({
+        isAdmin: getHostedCluster().isAdmin,
```

Comment:
Just a note that we should avoid `getHostedCluster()` as much as possible. It works only if code is executed within a cluster context (read: cluster frame). It's like a magic global variable which feels wrong.

Reply:
Alright